File: python/src/latticeag_herald/jkt.py
"""JWK thumbprint (RFC 7638) over the OKP public-key view used by the
adapter: jkt = b64u(SHA256(J({"crv":"Ed25519","kty":"OKP","x":<b64u>})))."""

import hashlib

from .jcs import canonicalize
from .b64u import b64u_encode


def jkt(public_key_b64u):
    return b64u_encode(hashlib.sha256(canonicalize(
        {"crv": "Ed25519", "kty": "OKP", "x": public_key_b64u})).digest())

File: packages/core/src/jkt.ts
/**
 * RFC 7638 JWK thumbprint for the card holder binding (§4):
 *   caller_jkt = b64u(SHA256(JCS({"crv":"Ed25519","kty":"OKP","x":<pk b64u>})))
 */

import { sha256Bytes } from "./digest.ts";
import { b64uEncode } from "./base64url.ts";
import { canonicalize } from "./jcs.ts";

export function callerJkt(publicKeyB64u: string): string {
  const jwk = { crv: "Ed25519", kty: "OKP", x: publicKeyB64u };
  return b64uEncode(sha256Bytes(canonicalize(jwk)));
}
